File: org.kevoree.modeling.microframework/src/test/java/org/kevoree/modeling/microframework/test/SliceTest.java
package org.kevoree.modeling.microframework.test;

import org.junit.Test;
import org.kevoree.modeling.api.ModelSlicer;
import org.kevoree.modeling.api.data.MemoryKDataBase;
import org.kevoree.modeling.microframework.test.cloud.CloudDimension;
import org.kevoree.modeling.microframework.test.cloud.CloudUniverse;
import org.kevoree.modeling.microframework.test.cloud.CloudView;
import org.kevoree.modeling.microframework.test.cloud.Node;

import java.util.Arrays;

/**
 * Created by duke on 10/16/14.
 */
public class SliceTest {

    @Test
    public void slideTest() {
        CloudUniverse universe = new CloudUniverse(new MemoryKDataBase());
        universe.newDimension((dimension0)->{
            CloudView time0 = dimension0.time(0l);
            Node root = time0.createNode();
            time0.setRoot(root);
            root.setName("root");

            Node n1 = time0.createNode();
            n1.setName("n1");

            Node n2 = time0.createNode();
            n2.setName("n2");

            root.addChildren(n1);
            root.addChildren(n2);

            ModelSlicer slicer = time0.createModelSlicer();
            slicer.slice(Arrays.asList(root), (seq) -> {
                System.err.println(seq);
            });
        });


    }

}

File: org.kevoree.modeling.microframework.typescript/src/test/ts/org/kevoree/modeling/microframework/test/SliceTest.ts
///<reference path="../../../../junit/Test.ts"/>
///<reference path="../../api/Callback.ts"/>
///<reference path="../../api/ModelSlicer.ts"/>
///<reference path="../../api/data/MemoryKDataBase.ts"/>
///<reference path="../../api/trace/TraceSequence.ts"/>
///<reference path="cloud/CloudDimension.ts"/>
///<reference path="cloud/CloudUniverse.ts"/>
///<reference path="cloud/CloudView.ts"/>
///<reference path="cloud/Node.ts"/>
///<reference path="../../../../../java/util/Arrays.ts"/>

class SliceTest {

  public slideTest(): void {
    var universe: CloudUniverse = new CloudUniverse(new MemoryKDataBase());
    universe.newDimension({on:function(dimension0: CloudDimension){
    var time0: CloudView = dimension0.time(0l);
    var root: Node = time0.createNode();
    time0.setRoot(root);
    root.setName("root");
    var n1: Node = time0.createNode();
    n1.setName("n1");
    var n2: Node = time0.createNode();
    n2.setName("n2");
    root.addChildren(n1);
    root.addChildren(n2);
    var slicer: ModelSlicer = time0.createModelSlicer();
}});
  }

}

